modules/ismp/clients/parachain/client/src/consensus.rs

```diff
@@ -131,7 +135,13 @@ where
 					{
 						let slot = Slot::decode(&mut &value[..])
 							.map_err(|e| Error::Custom(format!("Cannot slot: {e:?}")))?;
-						timestamp = Duration::from_millis(*slot * SLOT_DURATION).as_secs();
+
+						if para_value.slot_duration_type == ParaSlotDuration::Sync {
+							timestamp = Duration::from_millis(*slot * SYNC_SLOT_DURATION).as_secs();
+						} else if para_value.slot_duration_type == ParaSlotDuration::Async {
+							timestamp =
+								Duration::from_millis(*slot * ASYNC_SLOT_DURATION).as_secs();
+						}
```

Comment:
You're supposed to multiply the slot_number and the slot_duration to get the current timestamp